crates/meroctl/src/cli.rs

```diff
@@ -155,52 +148,51 @@ impl RootCommand {
-                        .await?;
-                Ok(Some(connection))
+            if let Some(conn) = config.get_connection(node, output).await? {
+                return Ok(conn);
             }
-            (None, Some(api_url)) => {
-                // Use specific API URL - check session cache first, then authenticate if needed
-                let connection =
-                    authenticate_with_session_cache(api_url, &api_url.to_string(), output).await?;
-                Ok(Some(connection))
-            }
-            (None, None) => {
-                // Try to use active node
-                let config = Config::load().await?;
-
-                if let Some(active_node_name) = &config.active_node {
-                    if let Some(conn) = config.get_connection(active_node_name, output).await? {
-                        return Ok(Some(conn));
-                    } else {
-                        bail!(
-                            "Active node '{}' not found. Please check your configuration.",
-                            active_node_name
-                        );
-                    }
-                }
 
-                // No active node set
-                Ok(None)
+            // Check if it's a local node at <home>/<node>
+            let config = load_config(&self.args.home, node).await?;
+            let multiaddr = fetch_multiaddr(&config)?;
+            let url = multiaddr_to_url(&multiaddr, "")?;
+
+            // Even local nodes might require authentication - use session cache for unregistered nodes
+            let connection =
+                authenticate_with_session_cache(&url, &format!("local node {}", node), output)
+                    .await?;
+            Ok(connection)
+        } else if let Some(api_url) = &self.args.api {
+            // Use specific API URL - check session cache first, then authenticate if needed
+            let connection =
+                authenticate_with_session_cache(api_url, &api_url.to_string(), output).await?;
+            Ok(connection)
+        } else {
+            // Try to use active node
+            let config = Config::load().await?;
+
+            if let Some(active_node_name) = &config.active_node {
+                if let Some(conn) = config.get_connection(active_node_name, output).await? {
+                    return Ok(conn);
+                } else {
+                    bail!(
+                        "Active node '{}' not found. Please check your configuration.",
+                        active_node_name
+                    );
+                }
             }
-            _ => Ok(None),
+
+            // No active node set - fall back to default localhost server
+            let default_url = "http://127.0.0.1:2528".parse()?;
+            let connection =
+                authenticate_with_session_cache(&default_url, "default localhost server", output)
+                    .await?;
```

Comment:
I believe this is where authentication happens, so we need the custom error here if the currently reported error doesn't describe the issue appropriately.